### PR TITLE
feat(image): add support for Docker CIS Benchmark

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
This PR adds support for [Docker CIS Benchmark](https://www.aquasec.com/cloud-native-academy/docker-container/docker-cis-benchmark/).

Docker CIS Benchmark has checks for vulnerabilities, misconfigurations and secrets. This PR supports only misconfigurations. Vulnerabilities and secrets will be supported later.

- [ ] Vulnerabilities
- [x] Misconfigurations
- [ ] Secrets

I'll update the documentation in another PR since we need to restructure the doc.

### Example

```
$ trivy image --compliance docker-cis my-image
2023-01-30T16:59:31.454+0900    INFO    Container image config scanners: ["config"]
2023-01-30T16:59:31.454+0900    INFO    Misconfiguration scanning is enabled
2023-01-30T16:59:31.511+0900    INFO    Detected config files: 1

Summary Report for compliance: CIS Docker Community Edition Benchmark v1.1.0
┌──────┬──────────┬────────────────────────────────────────────────────────────────────────────┬────────┬────────┐
│  ID  │ Severity │                                Control Name                                │ Status │ Issues │
├──────┼──────────┼────────────────────────────────────────────────────────────────────────────┼────────┼────────┤
│ 4.1  │   HIGH   │ Ensure a user for the container has been created                           │  FAIL  │   1    │
│ 4.2  │   HIGH   │ Ensure that containers use trusted base images (Manual)                    │   -    │   -    │
│ 4.3  │   HIGH   │ Ensure unnecessary packages are not installed in the container (Manual)    │   -    │   -    │
│ 4.4  │ CRITICAL │ Ensure images are scanned and rebuilt to include security patches (Manual) │   -    │   -    │
│ 4.5  │   LOW    │ Ensure Content trust for Docker is Enabled (Manual)                        │   -    │   -    │
│ 4.6  │   LOW    │ Ensure HEALTHCHECK instructions have been added to the container image     │  FAIL  │   1    │
│ 4.7  │   HIGH   │ Ensure update instructions are not use alone in the Dockerfile             │  PASS  │   0    │
│ 4.8  │   HIGH   │ Ensure setuid and setgid permissions are removed in the images (Manual)    │   -    │   -    │
│ 4.9  │   LOW    │ Ensure COPY is used instead of ADD in Dockerfile                           │  FAIL  │   1    │
│ 4.10 │ CRITICAL │ Ensure secrets are not stored in Dockerfiles (Manual)                      │   -    │   -    │
│ 4.11 │  MEDIUM  │ Ensure verified packages are only Installed (Manual)                       │   -    │   -    │
└──────┴──────────┴────────────────────────────────────────────────────────────────────────────┴────────┴────────┘
```

## Related issues
- https://github.com/aquasecurity/trivy/issues/2676

## Related PRs
- [x] https://github.com/aquasecurity/trivy/pull/3437
- [ ] https://github.com/aquasecurity/trivy/pull/3495
- [x] https://github.com/aquasecurity/defsec/pull/1144

Remove this section if you don't have related PRs.

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/commands/artifact/run.go [hunk 1]
```diff
@@ -476,6 +478,22 @@ func initScannerConfig(opts flag.Options, cacheClient cache.Cache) (ScannerConfi
...
+		// TODO: define image-config-scanners in the spec
+		if opts.Compliance.Spec.ID == "docker-cis" {
+			opts.Scanners = nil
+			opts.ImageConfigScanners = scanners
+		} else {
+			opts.Scanners = scanners
+			opts.ImageConfigScanners = nil
+		}
+	}
+
```
Comment (chen-keinan) [inline pkg/commands/artifact/run.go:496]: nit:
```suggestion
	if opts.Compliance.Spec.ID != "" {
		// set scanners types by spec
		scanners, err := opts.Compliance.Scanners()
		if err != nil {
			return ScannerConfig{}, types.ScanOptions{}, xerrors.Errorf("scanner error: %w", err)
		}
		opts.Scanners = scanners
		opts.ImageConfigScanners = nil
		// TODO: define image-config-scanners in the spec
		if opts.Compliance.Spec.ID == "docker-cis" {
			opts.Scanners = nil
			opts.ImageConfigScanners = scanners
		} 
	}

```

**Comment by chen-keinan** [comment]
@knqyf263 LGTM , I have added one `nit` comment its up to you

**Comment by AnaisUrlichs** [comment]
Should it be more highlighted in the documentation?

**Comment by knqyf263** [comment]
>Should it be more highlighted in the documentation?

Definitely! But there was no dedicated page for container images. I created it and mentioned Docker CIS Benchmark in https://github.com/aquasecurity/trivy/pull/3503.

**Comment by bijoy26** [comment]
Hi. Trivy currently supports CIS Docker `v1.1.0`. Is there any plan to incorporate other recent versions like `v1.6.0`?

**Pull request merged**